### PR TITLE
Cocoapods 1.9

### PR DESCRIPTION
**Pull request by djbe**
2 things of note:
- CDN sources, which allows us to get rid of the (still slow) step of downloading the full repo. We can now also cache the specs folder as it isn't as massive.
- `generate_multiple_pod_projects` and `incremental_installation`, which should reduce the amount of changes during `pod install/update`.

### CODE REVIEW

**Comment by SwiftGen-Eve** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">You should make sure that <code>Podfile.lock</code> and <code>Package.resolved</code> are changed in sync</td>
    </tr>
  </tbody>
</table>

Hey 👋 I'm Eve, the friendly bot watching over SwiftGen 🤖

Thanks a lot for your contribution!



---

Once you fix those tiny nitpickings above, we should be good to go! 🙌

ℹ️ _I will update this comment as you add new commits_

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by djbe** [comment]
Would you look at that, 4 seconds (without cache) 😆 🚗💨

**Review by AliSoftware — COMMENTED** [review]
We should also remove the deprecated `.swift-version` from the repo and use the `swift_versions` DSL attribute in the podspec instead

**Comment by djbe** [comment]
Uhm we can add the `swift_versions` bit, but some users seem to need that file to be there, see #555.

File: .circleci/config.yml [hunk 1]
```diff
@@ -143,29 +143,31 @@ jobs:
...
+          keys:
+            - cocoapods-{{ checksum "Podfile.lock" }}
+            - cocoapods-
       - run:
           name: Lint podspec
           command: bundle exec rake pod:lint
+      -  save_cache:
+          key: cocoapods-{{ checksum "Podfile.lock" }}
+          paths:
+            - ~/.cocoapods
```
Comment (AliSoftware) [inline .circleci/config.yml:156]: With CDN, is it really still worth it to cache the `~/.cocoapods` folder? From experience on other repos it doesn't really seem to be much necessary anymore.

We already commit the `Pods/` folder (and if we didn't, we could cache it) so it should be quite rare to have the need for the spec repo to be present and downloaded locally anyway, as it is now supposed to only be necessary during `pod update` (or if `pod install` needs to actually install pods that were missing in `Pods/` and were not at a version that was matching was was locked in `Podfile.lock`, which in our case should never happen on CI)

File: Gemfile [hunk 2]
```diff
@@ -4,19 +4,19 @@ source 'https://rubygems.org'
 
 # The bare minimum for building, e.g. in Homebrew
 group :build do
-	gem 'rake', '~> 12.3'
+	gem 'rake', '~> 13.0'
```
Comment (AliSoftware) [inline Gemfile:7]: 🎉 This should get rid of the security risk issue too 👍

**Review by AliSoftware — COMMENTED** [review]
Other than the fact that we should stop probably caching the CDN (and thus winning some more CI time by avoiding to restore it when not needed), LGTM

File: .circleci/config.yml (hunk 1 above)
Comment (AliSoftware) [reply .circleci/config.yml:156]: iow, if you remove those steps from CI, and as long as we still commit the `Pods/` folder and that `Podfile.lock` and `Pods/Manifest.lock` are properly in sync, I think `pod install` is now a no-op that shouldn't even require the CDN to be cloned at all.
Maybe I'm wrong so don't hesitate to make some tests to validate that if unsure.

**Review by AliSoftware — APPROVED** [review]
I thought the risk of using cache for specs repo was that cache restoration (download+unzip) and save (zip + upload) would risk taking more time than cloning the CDN. But after looking into it, I see that cache restoration only takes 3s. So we might as well keep it

**Pull request merged**